### PR TITLE
Refactoring API to be compatible with iOS version

### PR DESCRIPTION
**Pull request by mcassiano**

### CODE REVIEW

File: glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt [hunk 1]
```diff
@@ -24,10 +29,11 @@ class GlideImageComponent(
     }
 
     private val commands: Map<String, DynamicPropertyCommand> = mapOf(
-        WidthCommand.WIDTH_TYPE to WidthCommand(),
-        HeightCommand.HEIGHT_TYPE to HeightCommand(),
+        WIDTH_TYPE to WidthCommand(),
+        HEIGHT_TYPE to HeightCommand(),
+        SCALE_TYPE to ScaleTypePropertyCommand(),
```
Comment (rodrigo-mnz) [inline glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt:34]: Very nice :smile:, it's possible to add an adjustViewBounds options? or make it true by default. I had this problem while testing.

File: core/src/main/kotlin/br/com/concrete/yosef/api/property/size/HeightCommand.kt [hunk 2]
```diff
@@ -18,6 +20,21 @@ class HeightCommand : DynamicPropertyCommand {
     }
 
     override fun apply(view: View, dynamicProperty: DynamicProperty) {
-        view.layoutParams.height = dynamicProperty.value.toInt()
+        when {
```
Comment (rodrigo-mnz) [inline core/src/main/kotlin/br/com/concrete/yosef/api/property/size/HeightCommand.kt:23]: The first 'when' can be replaced by a normal 'if' or extract on a method, i think it'll be more readable.

File: core/src/main/kotlin/br/com/concrete/yosef/api/property/size/WidthCommand.kt [hunk 3]
```diff
@@ -15,9 +18,26 @@ class WidthCommand : DynamicPropertyCommand {
          * This constant documents which name is associated with the property
          */
         const val WIDTH_TYPE = "width"
+        const val MATCH = "match"
```
Comment (rodrigo-mnz) [inline core/src/main/kotlin/br/com/concrete/yosef/api/property/size/WidthCommand.kt:21]: This is great, let's just see if it's compatible with iOS :apple: .

**Review by rodrigo-mnz — CHANGES_REQUESTED** [review]

File: glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt (hunk 1 above)
Comment (mcassiano) [reply glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt:34]: Sure!

File: core/src/main/kotlin/br/com/concrete/yosef/api/property/size/WidthCommand.kt (hunk 3 above)
Comment (mcassiano) [reply core/src/main/kotlin/br/com/concrete/yosef/api/property/size/WidthCommand.kt:21]: I talked to an iOS engineer working on this and he said its fine, but I can hide this prop so its not misused now

File: core/src/main/kotlin/br/com/concrete/yosef/api/property/size/HeightCommand.kt (hunk 2 above)
Comment (mcassiano) [reply core/src/main/kotlin/br/com/concrete/yosef/api/property/size/HeightCommand.kt:23]: k! AS suggest it should be a when haha

File: glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt (hunk 1 above)
Comment (mcassiano) [reply glide/src/main/kotlin/br/com/concrete/yosef/glide/GlideImageComponent.kt:34]: I think you missed the ScaleTypePropertyCommand file. Theres a value options in which adjustViewBounds is on! Use scaleAspectFit option

**Pull request merged**